.github/workflows/maven-publish.yml

```diff
@@ -0,0 +1,42 @@
+name: Publish snapshots to maven
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [
+        main
+          1.*
+          2.*
+    ]
+
+jobs:
+  build-and-publish-snapshots:
+    strategy:
+      fail-fast: false
+      matrix:
+        jdk: [11, 17]
+        platform: ["ubuntu-latest", "windows-latest", "macos-latest"]
```

Comment:
I dont think we need to build on all platforms to publish to maven. Just ubuntu-latest works. Same with jdk.

Reply:
done